File: admin/src/supabaseService.js
import { createClient } from '@supabase/supabase-js'

const SUPABASE_URL = import.meta.env.VITE_SUPABASE_URL
const SUPABASE_ANON_KEY = import.meta.env.VITE_SUPABASE_ANON_KEY

export const supabase = createClient(SUPABASE_URL, SUPABASE_ANON_KEY)

export const supabaseService = {
  async getWartaTerbaru() {
    const { data, error } = await supabase
      .from('warta')
      .select('*')
      .eq('published', true)
      .order('created_at', { ascending: false })
      .limit(1)
      .single()
    
    if (error) throw error
    return data
  },

  async getAllWarta() {
    const { data, error } = await supabase
      .from('warta')
      .select('*')
      .eq('published', true)
      .order('created_at', { ascending: false })
    
    if (error) throw error
    return data
  },

  async createWarta(warta) {
    const { data, error } = await supabase
      .from('warta')
      .insert([warta])
      .select()
    
    if (error) throw error
    return data[0]
  },

  async updateWarta(id, warta) {
    const { data, error } = await supabase
      .from('warta')
      .update(warta)
      .eq('id', id)
      .select()
    
    if (error) throw error
    return data[0]
  },

  async deleteWarta(id) {
    const { error } = await supabase
      .from('warta')
      .delete()
      .eq('id', id)
    
    if (error) throw error
  },

  async uploadPDF(file) {
    const fileName = `${Date.now()}_${file.name}`
    const { data, error } = await supabase.storage
      .from('warta-pdf')
      .upload(fileName, file)
    
    if (error) throw error
    
    const { data: publicUrl } = supabase.storage
      .from('warta-pdf')
      .getPublicUrl(fileName)
    
    return publicUrl.publicUrl
  },

  async getJadwalIbadah() {
    const { data, error } = await supabase
      .from('jadwal_ibadah')
      .select('*')
      .order('tanggal', { ascending: true })
    
    if (error) throw error
    return data
  },

  async createJadwalIbadah(jadwal) {
    const { data, error } = await supabase
      .from('jadwal_ibadah')
      .insert([jadwal])
      .select()
    
    if (error) throw error
    return data[0]
  },

  async updateJadwalIbadah(id, jadwal) {
    const { data, error } = await supabase
      .from('jadwal_ibadah')
      .update(jadwal)
      .eq('id', id)
      .select()
    
    if (error) throw error
    return data[0]
  },

  async deleteJadwalIbadah(id) {
    const { error } = await supabase
      .from('jadwal_ibadah')
      .delete()
      .eq('id', id)
    
    if (error) throw error
  },

  async getPengumuman() {
    const { data, error } = await supabase
      .from('pengumuman')
      .select('*')
      .eq('published', true)
      .order('tanggal_mulai', { ascending: false })
    
    if (error) throw error
    return data
  },

  async createPengumuman(pengumuman) {
    const { data, error } = await supabase
      .from('pengumuman')
      .insert([pengumuman])
      .select()
    
    if (error) throw error
    return data[0]
  },

  async updatePengumuman(id, pengumuman) {
    const { data, error } = await supabase
      .from('pengumuman')
      .update(pengumuman)
      .eq('id', id)
      .select()
    
    if (error) throw error
    return data[0]
  },

  async deletePengumuman(id) {
    const { error } = await supabase
      .from('pengumuman')
      .delete()
      .eq('id', id)
    
    if (error) throw error
  },

  async getDoaPokok() {
    const { data, error } = await supabase
      .from('doa_pokok')
      .select('*')
      .order('periode_mulai', { ascending: false })
    
    if (error) throw error
    return data
  },

  async createDoaPokok(doa) {
    const { data, error } = await supabase
      .from('doa_pokok')
      .insert([doa])
      .select()
    
    if (error) throw error
    return data[0]
  },

  async updateDoaPokok(id, doa) {
    const { data, error } = await supabase
      .from('doa_pokok')
      .update(doa)
      .eq('id', id)
      .select()
    
    if (error) throw error
    return data[0]
  },

  async deleteDoaPokok(id) {
    const { error } = await supabase
      .from('doa_pokok')
      .delete()
      .eq('id', id)
    
    if (error) throw error
  },

  async getPersembahan() {
    const { data, error } = await supabase
      .from('persembahan')
      .select('*')
      .order('tanggal', { ascending: false })
    
    if (error) throw error
    return data
  },

  async createPersembahan(persembahan) {
    const { data, error } = await supabase
      .from('persembahan')
      .insert([persembahan])
      .select()
    
    if (error) throw error
    return data[0]
  },

  async updatePersembahan(id, persembahan) {
    const { data, error } = await supabase
      .from('persembahan')
      .update(persembahan)
      .eq('id', id)
      .select()
    
    if (error) throw error
    return data[0]
  },

  async deletePersembahan(id) {
    const { error } = await supabase
      .from('persembahan')
      .delete()
      .eq('id', id)
    
    if (error) throw error
  }
}

File: android/app/src/main/java/id/gkjst/ewarta/data/remote/SupabaseService.kt
package id.gkjst.ewarta.data.remote

import io.github.jan.supabase.client
import id.gkjst.ewarta.data.model.*
import javax.inject.Inject

class SupabaseService @Inject constructor() {

    suspend fun getWartaTerbaru(): Warta? {
        return try {
            val response = SupabaseConfig.client
                .from("warta")
                .select() {
                    filter {
                        eq("published", true)
                    }
                    order("created_at", ascending = false)
                    limit(1)
                }
                .decodeSingle<Warta>()
            response
        } catch (e: Exception) {
            null
        }
    }

    suspend fun getAllWarta(): List<Warta> {
        return try {
            SupabaseConfig.client
                .from("warta")
                .select() {
                    filter {
                        eq("published", true)
                    }
                    order("created_at", ascending = false)
                }
                .decodeList<Warta>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getJadwalIbadah(): List<JadwalIbadah> {
        return try {
            SupabaseConfig.client
                .from("jadwal_ibadah")
                .select() {
                    order("tanggal", ascending = true)
                }
                .decodeList<JadwalIbadah>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getJadwalSM(): List<JadwalSM> {
        return try {
            SupabaseConfig.client
                .from("jadwal_sm")
                .select() {
                    order("tanggal", ascending = true)
                }
                .decodeList<JadwalSM>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getPengumuman(): List<Pengumuman> {
        return try {
            SupabaseConfig.client
                .from("pengumuman")
                .select() {
                    filter {
                        eq("published", true)
                    }
                    order("tanggal_mulai", ascending = false)
                }
                .decodeList<Pengumuman>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getBacaanHarian(): List<BacaanHarian> {
        return try {
            SupabaseConfig.client
                .from("bacaan_harian")
                .select() {
                    order("tanggal", ascending = true)
                }
                .decodeList<BacaanHarian>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getUlangTahun(): List<UlangTahun> {
        return try {
            SupabaseConfig.client
                .from("ulang_tahun")
                .select()
                .decodeList<UlangTahun>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getDoaPokok(): List<DoaPokok> {
        return try {
            SupabaseConfig.client
                .from("doa_pokok")
                .select() {
                    order("periode_mulai", ascending = false)
                }
                .decodeList<DoaPokok>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getPersembahan(): List<Persembahan> {
        return try {
            SupabaseConfig.client
                .from("persembahan")
                .select() {
                    order("tanggal", ascending = false)
                }
                .decodeList<Persembahan>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun getWargaDoakan(): List<WargaDoakan> {
        return try {
            SupabaseConfig.client
                .from("warga_doakan")
                .select() {
                    filter {
                        eq("aktif", true)
                    }
                }
                .decodeList<WargaDoakan>()
        } catch (e: Exception) {
            emptyList()
        }
    }

    suspend fun uploadPDF(fileName: String, fileBytes: ByteArray): String? {
        return try {
            SupabaseConfig.client.storage
                .from("warta-pdf")
                .upload("$fileName-${System.currentTimeMillis()}.pdf", fileBytes)
            
            SupabaseConfig.client.storage
                .from("warta-pdf")
                .getPublicUrl("$fileName-${System.currentTimeMillis()}.pdf")
        } catch (e: Exception) {
            null
        }
    }
}
